app/package.json

```diff
@@ -15,7 +18,9 @@
     "type-check": "npx tsc --noEmit",
     "prepare": "cd .. && husky install app/.husky",
     "pre-commit": "lint-staged",
-    "generate": "npx @rtk-query/codegen-openapi openapi-config.ts"
+    "generate": "npx @rtk-query/codegen-openapi openapi-config.ts",
+    "check-node-version": "node check-node-version.mjs",
+    "preinstall": "npm run check-node-version"
```

Comment:
What's the purpose of "preinstall"?

Also, should we be running the "check-node-version" in front of the build scripts to prevent the build if the developer's node is out of date?